WireAPI/Sources/WireAPI/APIs/UserPropertiesAPI/UserPropertiesAPIV0.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class UserPropertiesAPIV0: UserPropertiesAPI, VersionedAPI {
+
+    let httpClient: HTTPClient
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    // MARK: - Fetch user property
+
+    func getProperty(forKey key: UserProperty.Key) async throws -> UserProperty {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/properties/\(key.rawValue)",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        switch key {
+        case .wireReceiptMode:
+            return try getProperty(response: response, payload: ReceiptModeResponseV0.self)
+        case .wireTypingIndicatorMode:
+            return try getProperty(response: response, payload: TypeIndicatorModeResponseV0.self)
+        case .labels:
+            return try getProperty(response: response, payload: LabelsResponseV0.self)
+        }
+    }
+
+    func getProperty<Payload: UserPropertiesResponseAPIV0>(
+        response: HTTPResponse,
+        payload: Payload.Type
+    ) throws -> UserProperty where Payload.APIModel == UserProperty {
+        try ResponseParser()
+            .success(code: .ok, type: payload)
+            .failure(code: .notFound, error: UserPropertiesAPIError.propertyNotFound)
```

Comment:
Not sure what you mean by that, could you be more specific ? Note that each failure case has a specific error e.g `UserPropertiesAPIError.propertyNotFound` which is an error specific to UserPropertiesAPI

Reply:
Oh good point that the error is specific to the `UserPropertiesAPI`. I hadn't considered that. I saw that `.failure(code: .notFound, error: UserPropertiesAPIError.propertyNotFound)` was repeated 4 times in this PR and thought it might happen a lot more. I meant we could have a convenience `notFoundFailure()` method so the calling is a little simpler:

```swift
 try ResponseParser()
            .success(code: .ok, type: payload)
            .notFoundFailure()
            .parse(response)
```

But considering that the error is API specific I don't think the change is worth it. It is better to be explicit. Please disregard my comment.